lib/bootloader_setup.pm

```diff
@@ -227,6 +228,22 @@ sub boot_grub_item {
     grub_key_enter;
 }
 
+sub get_scsi_id {
+    my $sid;
+    type_string_very_slow "devalias";
+    send_key 'ret';
+    save_screenshot;
+    my $file = './serial0';
+    open(my $fh, '<', "./serial0");
+    while (my $line = <$fh>) {
+        if ($line =~ /^disk/) {
+            $sid = ($line =~ m/scsi@(\d+)/)[0];
```

Comment:
open FILE, '<', $file_name or die $!; this is the standard to handle error
if ($line =~ /^disk/) 
should match start with 'disk : /pci' and have scsi in the string
once get sid, we should check if it's a number or not

Reply:
LGTM